### PR TITLE
kiwix: fix and update

### LINKED ISSUE #93447: kiwix fails to build
**Describe the bug**
Kiwix fails to build with a C++ code error.

**To Reproduce**
1. `export NIXPKGS_ALLOW_INSECURE=1`, because the package is marked as insecure,
2. `nix-env -iA nixos.kiwix`

**Expected behavior**
Kiwix builds.

**Additional context**
The error: 

```
In file included from ../../../src/common/pathTools.h:40,
                 from ../../../src/common/kiwix/reader.h:33,
                 from zimAccessor.cpp:59:
../../../src/common/stringTools.h:49:33: error: variable or field 'printStringInHexadecimal' declared void
   49 |   void printStringInHexadecimal(UnicodeString s);
      |                                 ^~~~~~~~~~~~~
../../../src/common/stringTools.h:49:33: error: 'UnicodeString' was not declared in this scope; did you mean 'icu_64::UnicodeString'?
   49 |   void printStringInHexadecimal(UnicodeString s);
      |                                 ^~~~~~~~~~~~~
      |                                 icu_64::UnicodeString
In file included from /nix/store/vpvz3710qq9jhcm3smb3mbk6j7lh47dl-icu4c-64.2-dev/include/unicode/translit.h:25,
                 from ../../../src/common/stringTools.h:23,
                 from ../../../src/common/pathTools.h:40,
                 from ../../../src/common/kiwix/reader.h:33,
                 from zimAccessor.cpp:59:
/nix/store/vpvz3710qq9jhcm3smb3mbk6j7lh47dl-icu4c-64.2-dev/include/unicode/unistr.h:289:20: note: 'icu_64::UnicodeString' declared here
  289 | class U_COMMON_API UnicodeString : public Replaceable
      |                    ^~~~~~~~~~~~~
make[4]: *** [Makefile:542: libZimAccessor_la-zimAccessor.lo] Error 1
make[4]: Leaving directory '/build/kiwix-0.9/src/components/zimAccessor'
make[3]: *** [Makefile:395: all] Error 2
make[3]: Leaving directory '/build/kiwix-0.9/src/components/zimAccessor'
make[2]: *** [Makefile:384: all-recursive] Error 1
make[2]: Leaving directory '/build/kiwix-0.9/src/components'
make[1]: *** [Makefile:396: all-recursive] Error 1
make[1]: Leaving directory '/build/kiwix-0.9/src'
make: *** [Makefile:456: all-recursive] Error 1
builder for '/nix/store/v2mvb8zdkbnsi7y4czcz6rnzd5bpdsw9-kiwix-0.9.drv' failed with exit code 2
error: build of '/nix/store/v2mvb8zdkbnsi7y4czcz6rnzd5bpdsw9-kiwix-0.9.drv' failed
```

Probably this needs to be solved for #35009. 

**Notify maintainers**

@robbinch

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.51, NixOS, 20.09pre234242.c71518e75bf (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(mario): `"nixos-20.09pre234242.c71518e75bf, home-manager"`
 - channels(root): `"nixos-20.09pre234242.c71518e75bf, home-manager-20.03"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: kiwix
```

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
Closes #93447, see also #117593

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/kiwix/default.nix [hunk 1]
```diff
@@ -1,109 +1,50 @@
...
-      --suffix PATH : ${aria2}/bin
+    wrapProgram $out/bin/kiwix-desktop \
+      --prefix PATH : ${lib.makeBinPath [ aria2 ]}
   '';
 
-  meta = {
+  meta = with lib; {
     description = "An offline reader for Web content";
     homepage = "https://kiwix.org";
     license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/kiwix/default.nix:46]: gpl3Only or gpl3Plus?

File: pkgs/applications/misc/kiwix/lib.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "034nk6l623v78clrs2d0k1vg69sbzrd8c0q79qiqmlkinck1nkxw";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    cmake
+    pkg-config
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/kiwix/lib.nix:30]: Do we really need all 4?

File: pkgs/development/libraries/mustache-hpp/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  installPhase = ''
+    mkdir -p $out/include
+    cp ../mustache.hpp $out/include
+  '';
+
+  meta = with lib; {
+    description = "Mustache text templates for modern C++";
+    homepage = "https://github.com/kainjow/Mustache";
+    license = licenses.boost;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mustache-hpp/default.nix:24]: Missing maintainer.

File: pkgs/development/libraries/zimlib/default.nix [hunk 4]
```diff
@@ -1,23 +1,56 @@
...
+    rev = version;
+    sha256 = "0iy0f1clhihq277x218ccx3mszgpr3h9l0by48b9ykr115nffw3s";
   };
 
-  buildInputs = [ lzma ];
+  nativeBuildInputs = [
+    meson
+    cmake
+    pkg-config
+    ninja
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zimlib/default.nix:27]: Do we really need all 4?

File: pkgs/development/libraries/zimlib/default.nix [hunk 5]
```diff
@@ -1,23 +1,56 @@
...
+  checkInputs = [
+    gtest
+  ];
+
+  doCheck = true;
 
   meta = with lib; {
     description = "Library for reading and writing ZIM files";
     homepage =  "https://www.openzim.org/wiki/Zimlib";
     license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zimlib/default.nix:52]: gpl2Only or gpl2Plus?

**Comment by risicle** [comment]
x86_64 linux:

```
builder for '/nix/store/pcnpl5pgcyzvq3k73m5ri6n2c594s73q-zimreader-0.92.drv' failed with exit code 2; last 10 log lines:
        |                      ^~~~
  main.cpp:84:20: error: 'class zim::File' has no member named 'good'
     84 |     if (!indexFile.good())
        |                    ^~~~
  make[2]: *** [Makefile:501: main.o] Error 1
  rm browsescreen.cpp zimcomp.cpp wikibits_js.cpp Wiki_png.cpp notfound.cpp linuxtag2009.cpp article.cpp redirect.cpp user_css.cpp search.cpp skin.cpp index.cpp number.cpp pager.cpp GFDL.cpp random.cpp searchresults.cpp main_css.cpp tntnet_png.cpp openzim_css.cpp browseresults.cpp browse.cpp openzim_skin.cpp monobookde_css.cpp ajax_js.cpp zimwp_css.cpp headbg_jpg.cpp monobook_skin.cpp searcharticles.cpp common_css.cpp commonPrint_css.cpp
  make[2]: Leaving directory '/build/source/zimreader/src'
  make[1]: *** [Makefile:348: all] Error 2
  make[1]: Leaving directory '/build/source/zimreader/src'
  make: *** [Makefile:449: all-recursive] Error 1
```

**Comment by ajs124** [comment]
@risicle, I'll just drop zimreader.  https://www.openzim.org/wiki/Zimreader

File: pkgs/development/libraries/mustache-hpp/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/mustache-hpp/default.nix:24]: Please do not resolve conversations which obviously are not resolved.

Comment (ajs124) [reply pkgs/development/libraries/mustache-hpp/default.nix:24]: Please stop "reviewing" my PRs.

**Pull request merged**